warp/src/raygun/mod.rs

```diff
@@ -1118,11 +1195,13 @@ pub trait RayGunStream: Sync + Send {
 
 #[async_trait::async_trait]
 pub trait RayGunEvents: Sync + Send {
+    #[deprecated(note = "Use `Conversation::send_event`")]
```

Comment:
I'm not sure why bother with deprecation. Do we have any external users and have we promised any API stability guarantees? I don't think we should unless we publish the crates on crates.io.

Reply:
It gives a chance to frontend implementation (specifically, uplink) to determine how they wish to proceed in the change while indicating to them that the functions are deprecated

Reply:
I really don't think that's needed here. The frontend is also us so if there is a doubt on the new API to be problematic for Uplink, we just make the changes there and see if it works as expected. If not, we rethink/redo the warp changes. Deprecation is meant for public APIs and it would make sense even across teams in the same organisation but that's not the case here. It's just us on both ends. :)

Reply:
While its not 100% needed to be deprecated, this type of change would be a major breaking change for uplink if they are removed right away. Keeping the api intact temporarily while they sort out how they wish to handle it internally since with this design it is not required to make repeated calls to `RayGun::get_conversation` but can make a single call and handle things from there based on any events emitted, etc. If this PR is decided on, the PR I would make on uplink would only do essential update to the code to ensure functionality while making note of the changes needed for somebody else to take care of that would fix their design on that platform. 

> The frontend is also us so if there is a doubt on the new API to be problematic for Uplink, we just make the changes there and see if it works as expected. If not, we rethink/redo the warp changes

Keep in mind, ideally warp itself may be available for anybody to implement, not just uplink (which there have been a couple of cases ive seen, though not sure of those implementation status). While we want to make sure the design make sense, we shouldnt have to go back and forth changing stuff warp just so it would fit an idea on one implementation (ie uplink) that may not be sound in other implementation. Always best to find the middle ground whenever possible. Just an FYI :)

Reply:
Thanks for explaining your arguments for this in detail. I appreciate that but I'm still not convinced this is worth it. Ideally we should never use deprecated API (not only in Uplink but anywhere) and not having to add "allow(deprecated)" that are easy to forget and ending up later using other deprecated API unintentionally.

> Keep in mind, ideally warp itself may be available for anybody to implement

That's not relevant to us. We don't expose a public API nor we make a claim about its stability.

More importantly. We are breaking the API big time in this PR itself. If we want to provide any stability, we should do that then. This half-way care of API break, is not going to help anyone.

Reply:
To sumarize, I really think we either guarantee API stability and hence always deprecate existing API and follow the semver rules, OR we don't. I don't see any reason to do both.